File: packages/compose/src/main/kotlin/ai/mobileagentos/compose/agent/AgentOutput.kt
package ai.mobileagentos.compose.agent

import ai.mobileagentos.compose.actions.AgentAction
import ai.mobileagentos.compose.memory.MemoryItem
import ai.mobileagentos.compose.privacy.PrivacyResult

data class AgentOutput(
    val response: String,
    val actions: List<AgentAction>,
    val readinessScore: Int,
    val session: AgentSession,
    val events: List<AgentEvent>,
    val privacyResult: PrivacyResult? = null,
    val retrievedMemories: List<MemoryItem>? = null
)

File: packages/swift/Sources/MobileAgentOS/Agent/AgentOutput.swift
import Foundation

/// Output from the agent runtime
public struct AgentOutput: Codable {
    public let response: String
    public let actions: [AgentAction]
    public let readinessScore: Int
    public let events: [AgentEvent]
    public let sessionId: String
    
    public init(
        response: String,
        actions: [AgentAction],
        readinessScore: Int,
        events: [AgentEvent],
        sessionId: String
    ) {
        self.response = response
        self.actions = actions
        self.readinessScore = readinessScore
        self.events = events
        self.sessionId = sessionId
    }
}